weasyl/favorite.py

```diff
@@ -254,7 +254,7 @@ def check(userid, submitid=None, charid=None, journalid=None):
         ], options="bool")
 
 
-def count(submitid):
+def count(submitid=None, journalid=None, charid=None):
     return d.execute(
-        "SELECT COUNT(*) FROM favorite WHERE targetid = %i AND type = 's'",
-        [submitid], options=['element'])
+        "SELECT COUNT(*) FROM favorite WHERE targetid = %i AND type = '%s'",
+        [submitid, 's' if submitid else 'f' if charid else 'j'], options=['element'])
```

Comment:
This should be refactored:

1: I'm not sure it does the right thing. It looks like it will always use submitid even if it's None.
2: The chained ifs strike me as a bit of an anti-pattern.
3: It should probably be an error to specify other than exactly one of submitid, journalid, or charid.

Maybe this should be rewritten to take a type and an id. Then ideally the type should not be a single character string (because we want to move away from single character settings) but an enum. Enums, though, are cheap.

Python likes short strings for constant symbols so how about taking an `id` and a `type='submission'` and making it an error if `type` is anything other than `'submission', 'journal'` or `'character'`?

Also should have a doc string for the function.